packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/direct-edit/DiagramDirectEditInput.tsx

```diff
@@ -88,6 +94,15 @@ export const DiagramDirectEditInput = ({ labelId, editingKey, onClose, transform
 
   const [renameElement, { data: editLabelData, error: editLabelError }] = useMutation(editLabelMutationOp);
 
+  useEffect(() => {
+    return () => {
+      let elementToFocus: HTMLDivElement | null = (parentElementRef.current?.parentNode as HTMLDivElement) ?? null;
+      if (elementToFocus) {
+        elementToFocus.focus();
+      }
+    };
```

Comment:
You need a ref to the parent element to force the focus on it.
You may want to support edges too; thus, `HTMLDivElement` may not be appropriate here.